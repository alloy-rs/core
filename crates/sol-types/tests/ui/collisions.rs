use alloy_sol_types::sol;

// https://github.com/alloy-rs/core/issues/729

sol! {
    error func_2093253501(bytes);
    error transfer(address,uint256);

    function func_2093253501(bytes);
    function transfer(address,uint256);

    error BlazingIt4490597615();

    contract A {
        error func_2093253501(bytes);
        error transfer(address,uint256);

        function func_2093253501(bytes);
        function transfer(address,uint256);

        error BlazingIt4490597615();
    }
}

// This is OK.
mod namespaced {
    use alloy_sol_types::sol;

    sol! {
        function func_2093253501(bytes);

        contract B {
            function transfer(address,uint256);
        }
    }
}

fn main() {}
