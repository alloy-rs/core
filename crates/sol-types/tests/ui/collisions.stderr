error: function selector `0xa9059cbb` collides with `func_2093253501`
  --> tests/ui/collisions.rs:10:14
   |
10 |     function transfer(address,uint256);
   |              ^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/collisions.rs:9:14
   |
9  |     function func_2093253501(bytes);
   |              ^^^^^^^^^^^^^^^

error: error selector `0x00000000` is reserved
  --> tests/ui/collisions.rs:12:11
   |
12 |     error BlazingIt4490597615();
   |           ^^^^^^^^^^^^^^^^^^^

error: function selector `0xa9059cbb` collides with `func_2093253501`
  --> tests/ui/collisions.rs:19:18
   |
19 |         function transfer(address,uint256);
   |                  ^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/collisions.rs:18:18
   |
18 |         function func_2093253501(bytes);
   |                  ^^^^^^^^^^^^^^^

error: error selector `0x00000000` is reserved
  --> tests/ui/collisions.rs:21:15
   |
21 |         error BlazingIt4490597615();
   |               ^^^^^^^^^^^^^^^^^^^
